modules/exploits/windows/local/ms16_032_secondary_logon_handle_privesc.rb

```diff
@@ -0,0 +1,519 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/payload_generator'
+require 'rex'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Exploit::Powershell
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'MS16-032 Secondary Logon Handle Privilege Escalation',
+      'Description'   => %q{
+This module exploits the lack of sanitization of standard handles in Windows' Secondary Logon Service.  The vulnerability is known to affect versions of Windows 7-10 and 2k8-2k12 32 and 64 bit.  This module will only work against those versions of Windows with Powershell 2.0 or later and systems with two or more CPU cores. Works about 75% of the time, if module fails, just re-run.
```

Comment:
Hola, I am sorry to be late to the party but I wanted to note that I fixed the reliability issue, looks to be 99.9999% now and no more false success. Also, even though there is still the race condition, getting a shell should be almost instant.

Github:
https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/Invoke-MS16-032.ps1

![cnmng0kwiaashxo](https://cloud.githubusercontent.com/assets/10997340/16861980/c3b5364a-4a3d-11e6-8d0d-8c00ca6e9222.jpg)